types/coin.go

```diff
@@ -270,6 +282,10 @@ func (coins Coins) Add(coinsB ...Coin) Coins {
 // denomination and addition only occurs when the denominations match, otherwise
 // the coin is simply added to the sum assuming it's not zero.
 func (coins Coins) safeAdd(coinsB Coins) Coins {
+	if !coinsB.isSorted() {
+		panic("coinsB array is not sorted")
```

Comment:
It might be hard to determine what `coinsB` is when running into this error since it will be quite far down the stack. Maybe return the actual coin set?

Reply:
Thanks for review. We can create a new array and sort it. I wanted to avoid sorting in place to not change the argument - when calling `coinsA.Add(coinsB)`, it could be surprising for a user that we are changing the argument (`coinsB`).

Reply:
How about `"Wrong arguments: coins must be sorted"`?

Reply:
updated